internal/repos/repo.go

```diff
@@ -91,3 +93,28 @@ func generateRepoRule(repo repo) bf.Expr {
 	}
 	return rules.NewRule("go_repository", attrs)
 }
+
+// FindExternalRepo attempts to locate the directory where Bazel has fetched
+// the external repository with the given name. An error is returned if the
+// repository directory cannot be located.
+func FindExternalRepo(repoRoot, name string) (string, error) {
+	// See https://docs.bazel.build/versions/master/output_directories.html
+	// for documentation on Bazel directory layout.
+	// We expect the bazel-out symlink in the workspace root directory to point to
+	// <output-base>/execroot/<workspace-name>/bazel-out
+	// We expect the external repository to be checked out at
+	// <output-base>/external/<name>
+	externalPath := strings.Join([]string{repoRoot, "bazel-out", "..", "..", "..", "external", name}, string(os.PathSeparator))
```

Comment:
I know some people that modify this name by setting --symlink_prefix in their bazelrc, and also if the build is failing the convenience symlinks might not be made
`bazel info output_base` would be more correct, but has the cost of invoking bazel...

Reply:
--symlink_prefix doesn't change the name of bazel-out. This is mentioned in the documentation, so it's intentional, hopefully stable behavior. Annoyingly inconsistent, but useful in this case.

I expect this function may be called frequently (at least once per external repo on each fix/update run), so I'd like to avoid invoking bazel here if at all possible.